### ISSUE TITLE
Node positions not being according with specs

### ISSUE THREAD

**Issue by dpordomingo** [bblfsh/java-driver#27]
It is not retrieved `start_position.col` field nor `end_position` as documented by https://doc.bblf.sh/uast/specification.html
> _[...] it is guaranteed that nodes in a UAST either have no position attached or they have a position with valid offset, line and col. [...]_
> _End position, if present in a token node, is the position of the last character of the token in the original source code._

![pasted image at 2017_06_27 09_52 am](https://user-images.githubusercontent.com/2437584/27594993-af4877c8-5b5b-11e7-909a-8725014e703d.png)

example:
```
public class HelloWorld
{
 public static void main(String[] args)
 {
  System.out.println("Hello world!");
 }
}
```
client: [client-python](https://github.com/bblfsh/client-python)
It is retrieved the following positions for `out` Node
```
start_position {
    offset: 78
    line: 5
}
```

**Comment by abeaumont** [comment]
Thanks for the report. The UAST seems to have a `col` set always to 0 and no `end_position`. Somewhere in the serialization process it seems that a 0 `col` is converted to no value and a no `end_position` is converted to an invalid `end_position`.

**Comment by smola** [comment]
Note that empty position is a valid position, it's the zero value and means _no position_. The spec does not mandate that end position is present. In fact, it does not mandate that start position is present either:

>  Nodes with defined token SHOULD have, at least, start position, which correspond to the position of the first character of the token in the original source code. End position, if present in a token node, is the position of the last character of the token in the original source code.

The missing col is indeed a bug. It needs to be calculated with `driver.TransformationUASTParser` + `driver.FillLineColFromOffset` from `bblfsh/sdk`. That needs to wrap the parser in the `driver/normalizer/parser.go` (`UASTParserBuilder`).

**Comment by juanjux** [comment]
I've moved the endposition part as enhancement request #30. As for the column numbers, PR #31 should fix it when merged.

**Comment by abeaumont** [comment]
This should be fixed now thanks to #41
